src/Symfony/Bridge/PhpUnit/SymfonyTestsListenerTrait.php

```diff
@@ -0,0 +1,281 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\PhpUnit;
+
+use Doctrine\Common\Annotations\AnnotationRegistry;
+use PHPUnit\Framework\AssertionFailedError;
+use PHPUnit\Framework\TestCase;
+use PHPUnit\Framework\TestSuite;
+use PHPUnit\Framework\Warning;
+use PHPUnit\Runner\BaseTestRunner;
+use PHPUnit\Util\Blacklist;
+use PHPUnit\Util\Test;
+
+/**
+ * Collects and replays skipped tests.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @internal
+ */
+trait SymfonyTestsListenerTrait
+{
+    private static $globallyEnabled = false;
+    private $state = -1;
+    private $skippedFile = false;
+    private $wasSkipped = array();
+    private $isSkipped = array();
+    private $expectedDeprecations = array();
+    private $gatheredDeprecations = array();
+    private $previousErrorHandler;
+    private $testsWithWarnings;
+
+    /**
+     * @param array $mockedNamespaces List of namespaces, indexed by mocked features (time-sensitive or dns-sensitive)
+     */
+    public function __construct(array $mockedNamespaces = array())
+    {
+        if (!class_exists(Blacklist::class)) {
+            \PHPUnit_Util_Blacklist::$blacklistedClassNames['\Symfony\Bridge\PhpUnit\DeprecationErrorHandler'] = 1;
+            \PHPUnit_Util_Blacklist::$blacklistedClassNames['\Symfony\Bridge\PhpUnit\SymfonyTestsListener'] = 1;
+        } else {
+            Blacklist::$blacklistedClassNames['\Symfony\Bridge\PhpUnit\DeprecationErrorHandler'] = 1;
+            Blacklist::$blacklistedClassNames['\Symfony\Bridge\PhpUnit\SymfonyTestsListener'] = 1;
```

Comment:
I fact I fogot that the bridge needs to be PHP 5.3 compatible, so I removed all the `::class`